flink-runtime/src/main/java/org/apache/flink/runtime/externalresource/ExternalResourceUtils.java

```diff
@@ -0,0 +1,220 @@
+		LOG.info("Enabled external resources: {}", resourceSet);
+
+		if (resourceSet.isEmpty()) {
+			return Collections.emptyMap();
+		}
+
+		final Map<String, Long> externalResourceAmountMap = new HashMap<>();
+		for (String resourceName: resourceSet) {
+			final ConfigOption<Long> amountOption =
+				key(ExternalResourceOptions.keyWithResourceNameAndSuffix(resourceName, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX))
+					.longType()
+					.noDefaultValue();
+			final Optional<Long> amount = config.getOptional(amountOption);
+			if (!amount.isPresent() || amount.get() <= 0) {
+				LOG.warn("The amount of the {} should be configured and the value should be positive. Will ignore that resource.", resourceName);
+				continue;
+			} else {
+				externalResourceAmountMap.put(resourceName, amount.get());
+			}
+		}
+
+		return externalResourceAmountMap;
+	}
+
+	/**
+	 * Instantiate the {@link ExternalResourceDriver}s for all of enabled external resources. {@link ExternalResourceDriver}s
+	 * are mapped by its resource name.
+	 */
+	public static Map<String, ExternalResourceDriver> externalResourceDriversFromConfig(Configuration config, PluginManager pluginManager) {
+		final Set<String> resourceSet = getExternalResourceSet(config);
+		LOG.info("Enabled external resources: {}", resourceSet);
+
+		if (resourceSet.isEmpty()) {
+			return Collections.emptyMap();
+		}
+
+		final Iterator<ExternalResourceDriverFactory> factoryIterator = pluginManager.load(ExternalResourceDriverFactory.class);
+		final Map<String, ExternalResourceDriverFactory> externalResourceFactories = new HashMap<>();
+		factoryIterator.forEachRemaining(
+			externalResourceDriverFactory -> {
+				externalResourceFactories.put(externalResourceDriverFactory.getClass().getName(), externalResourceDriverFactory);
+			});
+
+		final Map<String, ExternalResourceDriver> externalResourceDrivers = new HashMap<>();
+		for (String resourceName: resourceSet) {
+			final ConfigOption<String> driverClassOption =
+				key(ExternalResourceOptions.keyWithResourceNameAndSuffix(resourceName, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX))
+					.stringType()
+					.noDefaultValue();
+			final String driverFactoryClassName = config.getString(driverClassOption);
+			if (StringUtils.isNullOrWhitespaceOnly(driverFactoryClassName)) {
+				LOG.warn("Could not found driver class name for {}. Please make sure {} is configured.",
+					resourceName, driverClassOption.key());
+				continue;
+			}
+
+			if (externalResourceFactories.containsKey(driverFactoryClassName)) {
+				DelegatingConfiguration delegatingConfiguration =
+					new DelegatingConfiguration(config, ExternalResourceOptions.keyWithResourceNameAndSuffix(resourceName, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_PARAM_SUFFIX));
+				try {
+					externalResourceDrivers.put(resourceName, externalResourceFactories.get(driverFactoryClassName).createExternalResourceDriver(delegatingConfiguration));
```

Comment:
Duplicated access to `externalResourceFactories`.
Could replace with first `externalResourceFactories.get(driverFactoryClassName)` then check whether the value is null.